src/rendering/RenderUtil.cc

```diff
@@ -1009,6 +1141,15 @@ void RenderUtilPrivate::RemoveRenderingEntities(
         this->removeEntities[_entity] = _info.iterations;
         return true;
       });
+
+  // collisions
+  _ecm.EachRemoved<components::Collision>(
+    [&](const Entity &_entity, const components::Collision *)->bool
+      {
+        this->removeEntities[_entity] = _info.iterations;
+        this->viewingCollisions.erase(_entity);
```

Comment:
do you also need to update `entityCollisions` and `linkToCollisionEntities`, and probably also `modelToLinkEntities` when links are removed? hmm there's a bit of maintenance that need to be done to keep these maps update-to-date.

Reply:
Done in 85223984950c065ecaf711efd05c13048fd0a547. Since only a model or a light can be removed from the gui (not a link or a collision), I didn't have to iterate through the maps to keep them updated.